alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -13,7 +11,9 @@ RunnableObject::RunnableObject(IAlicaWorldModel* wm, const std::string& name)
         : _name(name)
         , _engine(nullptr)
         , _configuration(nullptr)
-        , _flags(static_cast<uint8_t>(Flags::TRACING_ENABLED))
+        , _tracingType(TracingType::DEFAULT)
+        , _runTraced(false)
+        , _initExecuted(false)
```

Comment:
why 2 bools instead of Enum? both bools are related to tracing, right? but 2 bools is not as scalable, and bugs can make weird states (like `!_initExecuted and _runTraced`). maybe i'm missing something. please document both vairables

Reply:
these bools track different things. initExecuted is for when the init of the plan/behaviour is executed (it could be skipped if we are not in context of the current running plan) & runTraced is to ensure that the run method is traced only once instead of on every call